app/src/main/java/org/stacks/app/domain/GenerateNewIdentityKeys.kt

```diff
@@ -0,0 +1,56 @@
+package org.stacks.app.domain
+
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.withContext
+import org.blockstack.android.sdk.model.BlockstackIdentity
+import org.blockstack.android.sdk.toBtcAddress
+import org.kethereum.bip32.generateChildKey
+import org.kethereum.bip32.toKey
+import org.kethereum.bip39.model.MnemonicWords
+import org.kethereum.bip39.toSeed
+import org.kethereum.extensions.toHexStringNoPrefix
+import org.kethereum.model.ECKeyPair
+import org.komputing.kbip44.BIP44Element
+import org.stacks.app.data.EncryptedPreferencesIdentityRepository
+import org.stacks.app.data.EncryptedPreferencesSecretKeyRepository
+import timber.log.Timber
+import javax.inject.Inject
+
+class GenerateNewIdentityKeys
+@Inject constructor(
+    private val identityRepository: EncryptedPreferencesIdentityRepository,
+    private val secretKeyRepository: EncryptedPreferencesSecretKeyRepository
+) {
+
+    //TODO:
+    // test size variants
+    fun generate(): Flow<ECKeyPair> = identityRepository.observe()
```

Comment:
Why is this a flow, if we're just taking 1? Should be a normal suspend, with no stream, and it will look much simpler.